### DISCUSSION TITLE
is it safe to upgrade to v15 with latest expo

### DISCUSSION THREAD

**Discussion by ScreamZ** [invertase/react-native-firebase#6607] [General]
> ALL iOS: firebase-ios-sdk now requires use_frameworks! in your Podfile and Xcode 13.3+. Note that use_frameworks is not yet compatible with Hermes, Flipper, React Native New Architecture, or react-native 0.69.0. Each of these is being worked on (follow https://github.com/facebook/react-native/pull/34011, https://github.com/facebook/react-native/pull/34030) but for now you need react-native 0.68.2 or below, and you must disable hermes and flipper. Expo users should use [expo-build-properties](https://docs.expo.dev/versions/v45.0.0/sdk/build-properties/#pluginconfigtypeios) + Expo SDK45 to turn on use_frameworks in dynamic mode.

Is is safe now to upgrade to v15 with expo ?

**Comment by mikehardy** [comment]
How did it go when you tried it?
We have several people using it, as evidenced by the guides to turning on use_frameworks with expo, but of course any app may use dependencies that cannot tolerate use_frameworks.

You try it with your app and only then you will know

**Reply by ScreamZ** [reply]
Could we agree that as long as it build it works ? 😅 Or it may happen some unexpected side effects even if it build ?

Thanks :-)

**Reply by mikehardy** [reply]
If it builds it *probably* works, but all you know for sure is that it builds, your testing harness (or manual testing) is the only way to know for sure, but...building is a good sign, that is the normal probably with the switch to use_frameworks -- total compilation failure on iOS

**Reply by ScreamZ** [reply]
Alright, keeping my investigations I can currently state that
- v15+ works on android build with Expo
- v14 works on iOS with Expo
- v15 does not build without use framework on iOS
```
[
  "expo-build-properties",
  {
    "ios": {
      "useFrameworks": "static"
    }
  }
]
```
- v15 build with useframework on iOS, works in expo development client (in dev mode) but crash when you launch the standalone build on start

I'll try to isolate a minimum reproducible example with only RN firebase if I can to see if it's related to this or another library

Maybe you've some intel already @mikehardy ?

**Reply by mikehardy** [reply]
I don't understand? `use_frameworks! :linkage => :static` is a strict, documented requirement for react-native-firebase v15 on iOS, so that's not novel as a result.

This though:

> v15 build with useframework on iOS, works in expo development client (in dev mode) but crash when you launch the standalone build on start

It appears you need to either start it with Xcode or look at Console.app while device is plugged in to development machine to understand the stack trace. I have on special intel there, I know there are people using Expo and react-native-firebase v15 though, so I *suspect* (without proof yet) that is another library interacting

**Reply by tyrauber** [reply]
@mikehardy, I believe this is the same issue I experienced. I created a ticket at [expo/expo](https://github.com/expo/expo/issues/19865). This is an issue with expo-dev-client, useFrameworks 'static' and jsEngine 'hermes'.

**Reply by ScreamZ** [reply]
@tyrauber I put the link here for remind https://forums.expo.dev/t/expo-46-eas-react-native-firebase-and-hermes/67396

**Reply by tyrauber** [reply]
The issue at [expo/expo](https://github.com/expo/expo/issues/19865) was closed as [an upstream issue](https://github.com/facebook/react-native/issues/34344) which needs to be resolved by the react-native team. Until then, we need to lock react-native-firebase to v14.11.0, or don't use hermes.
